app/services/repository_actions/duplicate_cell.rb

```diff
@@ -0,0 +1,110 @@
+# frozen_string_literal: true
+
+module RepositoryActions
+  class DuplicateCell
+    def initialize(cell, new_row, user)
+      @cell    = cell
+      @new_row = new_row
+      @user    = user
+      @team    = user.current_team
```

Comment:
@ZmagoD sorry for being pain in the neck. I still have one comment - I wouldn't count on `user.current_team` to be the correct team into which to clone the data. If we are certain that `@new_row` is already persisted, we should retrieve the `@team` from this new row (`@new_row.repository.team`), not from the user.

I know it might be more DB querying than checking the user's current team, but it is definitely more reusable if we will ever call this service from some other piece of code in the future (e.g. from the API, where user's current team might be anything).